x/bounty/keeper/finding.go

```diff
@@ -124,3 +131,64 @@ func BytesToUint64s(list []byte) ([]uint64, error) {
 	}
 	return r64, nil
 }
+
+func CheckPlainText(pubKey *ecies.PublicKey, plainText string, anyData *codectypes.Any) (bool, error) {
+	var randBytes []byte
+	var encryptData []byte
+
+	if anyData.TypeUrl == "/shentu.bounty.v1.EciesEncryptedDesc" {
+		var encrypted types.EciesEncryptedDesc
+		err := proto.Unmarshal(anyData.GetValue(), &encrypted)
+		if err != nil {
+			return false, err
+		}
+
+		if len(encrypted.FindingDesc) < cli.RandBytesLen {
+			return false, types.ErrFindingEncryptedDataInvalid
+		}
+		randBytesStart := len(encrypted.FindingDesc) - cli.RandBytesLen
+		encryptData = encrypted.FindingDesc[:randBytesStart]
+		randBytes = encrypted.FindingDesc[randBytesStart:]
+
+	} else if anyData.TypeUrl == "/shentu.bounty.v1.EciesEncryptedPoc" {
+		var encrypted types.EciesEncryptedPoc
+		err := proto.Unmarshal(anyData.GetValue(), &encrypted)
+		if err != nil {
+			return false, err
+		}
+
+		if len(encrypted.FindingPoc) < cli.RandBytesLen {
+			return false, types.ErrFindingEncryptedDataInvalid
+		}
+		randBytesStart := len(encrypted.FindingPoc) - cli.RandBytesLen
+		encryptData = encrypted.FindingPoc[:randBytesStart]
+		randBytes = encrypted.FindingPoc[randBytesStart:]
+	} else if anyData.TypeUrl == "/shentu.bounty.v1.EciesEncryptedComment" {
```

Comment:
I'm not too sure if these should all be separated into different fields. Can we merge them into an interface? Like `content` interface that can contain both encrypted and decrypted description, PoC and `comment` interface that can contain both encrypted and decrypted comments

Reply:
Now the findingPoc field can only be encrytped data or plaintext. Encrypted data and plaintext used one field.